runtime/sema/type.go

```diff
@@ -3411,6 +3411,7 @@ func init() {
 			HashAlgorithmType,
 			StorageCapabilityControllerType,
 			AccountCapabilityControllerType,
+			&InclusiveRangeType{},
```

Comment:
I guess this is not needed here, because `InclusiveRange` is a dynamically created type (just like arrays, dictionaries, etc.)

Reply:
It is needed right. If I don't declare it, you can't refer to it in the code.

```
let a : InclusiveRange<Int> = InclusiveRange(1, 2)
```

It is different from arrays, dictionary because you don't refer to it in the code.

```
let a : Dictionary<Int, String> = ...
```

Reply:
ah ok. I think then the `checkParameterizedTypeIsInstantiated` must be recursive.
i.e: It should also look for the types nested within other types.

For e.g: An array of `InclusiveRange` (without the type parameter), can be defined without any errors.
```cadence
let r: [InclusiveRange] = []    // no errors
```

Reply:
Maybe fix that in a follow-up PR - would make it easy to review (this PR is already big)